pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,429 @@
+        # Inform the item that the mouse is(not) hovering over it
+        if self.mouseHovering is hover:
+            return
+        self.mouseHovering = hover
+        if hover:
+            self.currentBrush = self.hoverBrush
+            self.currentPen = self.hoverPen
+        else:
+            self.currentBrush = self.brush
+            self.currentPen = self.pen
+        self.update()
 
     def hoverEvent(self, ev):
-        if self.movable:
-            ev.acceptDrags(QtCore.Qt.LeftButton)
+        if self.movable and (not ev.isExit()) and ev.acceptDrags(QtCore.Qt.LeftButton):
+            self.setMouseHover(True)
+        else:
+            self.setMouseHover(False)
+
+    def viewTransformChanged(self):
+        GraphicsObject.viewTransformChanged(self)
+        self._shape = None  # invalidate shape, recompute later if requested.
+        self.update()
+
+    def pos(self):
+        """Provides the current position of the TargetItem
+
+        Returns
+        -------
+        tuple
+            ``(x, y)`` position of the TargetItem
+        """
+        return self._pos
+
+    def label(self):
+        """Provides the TargetLabel if it exists
+
+        Returns
+        -------
+        TargetLabel or None
+            If a TargetLabel exists for this TargetItem, return that, otherwise
+            return None
+        """
+        return self._label
+
+    def setLabel(self, text=None, labelOpts=None):
+        """Method to call to enable or disable the TargetLabel for displaying text
+
+        Parameters
+        ----------
+        text : Callable or str, optional
+            Details how to format the text, by default None
+            If None, do not show any text next to the TargetItem
+            If Callable, then the label will display the result of ``text(x, y)``
+            If a fromatted string, then the output of ``text.format(x, y)`` will be
+            displayed
+            If a non-formatted string, then the text label will display ``text``, by
+            default None
+        labelOpts : dictionary, optional
+            These arguments are passed on to :class:`~pyqtgraph.TextItem`
+        """
+        if text is None and self._label is not None:
+            # remove the label if it's already added
+            if self._label.scene() is not None:
+                self._label.scene().removeItem(self._label)
+            self._label = None
+        else:
+            # convert None entry for text to False
+            text = False if text is None else text
+
+            # provide default text if text is True
+            if isinstance(text, bool):
+                # convert to default value or empty string
+                text = "x = {: .3n}\ny = {: .3n}" if text else ""
+
+            # if there is no label, add a label
+            if self._label is None:
+                labelOpts = {} if labelOpts is None else labelOpts
+                self._label = TargetLabel(self, text=text, **labelOpts)
```

Comment:
I pared down one of the example uses to a minimal `pg.TargetItem()`, and it blows up here:
```
Traceback (most recent call last):
  File "examples/InfiniteLine.py", line 35, in <module>
    targetItem1 = pg.TargetItem()
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 120, in __init__
    self.setLabel(label, labelOpts)
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 327, in setLabel
    self._label = TargetLabel(self, text=text, **labelOpts)
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 375, in __init__
    self.setFormat(text)
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 404, in setFormat
    if parsed[0][1] is not None:
IndexError: list index out of range
```

Maybe this method needs another top-level if case for `text is None and labelOpts is None`? We could also fix it inside of TargetLabel, and in fact we might want both.

Reply:
I made a modification where a text label is not generated if test resolves `False`, so that should cover, `""`, `None` and `False`.

In `TargetLabel`, instead of defaulting to `None`, I think I'll default to the empty string, just like `TextItem`

Reply:
I think this is resolved sufficiently; I now check that the `parsed` list at least exists; which should be sufficient for our use-case.